internal/v4/auth.go

```diff
@@ -76,41 +81,71 @@ func (h *ReqHandler) authorize(req *http.Request, acl []string, alwaysAuth bool,
 	return authorization{}, httpbakery.NewDischargeRequiredErrorForRequest(m, cookiePath, verr, req)
 }
 
-func (h *ReqHandler) checkTerms(id *router.ResolvedURL, req *http.Request) error {
-	user, passwd, err := parseCredentials(req)
-	if err == nil {
-		if user != h.handler.config.AuthUsername || passwd != h.handler.config.AuthPassword {
-			return errgo.WithCausef(nil, params.ErrUnauthorized, "invalid user name or password")
-		}
-		return nil
+func (h *ReqHandler) authorizeWithTerms(req *http.Request, entityId *router.ResolvedURL) (authorization, error) {
```

Comment:
authorizeEntityAndTerms
?

(usually the "With" qualifier in a name means that we're passing something extra into the method)

Also, please add a doc comment.